net/xia/ppal_ether/main.c

```diff
@@ -350,7 +419,9 @@ static int main_delroute(struct xip_ppal_ctx *ctx,
 		goto unlock_bucket;
 	}
 
+	mether->cached_hdr.hh_len = 0;
```

Comment:
Why do we need this line?

Reply:
Because as soon as the request to delete a ether-xid entry is called, even if a process is in the function `interface_neigh_output` and is trying to access the cached header it won't be able to send the packet while we are going through the process of deleting the fxid entry.

Reply:
I think we should drop this line because all we're trying to accomplish here is to reduce the time that packets will follow the removed XID, and the RCU mechanism is already in place to deal with these transitions. It's okay for a couple of packets to use the removed XID during the transition.

The reason this line is troubling is that it's editing a data structure that could potentially be being used by another thread without any synchronization mechanism in place.